### PR TITLE
Fix leak of generated classes caused by using a shared loader

### PR DESCRIPTION
**Pull request by plesner**
It's a hard problem to write a regression test for, particularly because java is reluctant to clean up class loaders. However, the old code definitely kept all generated classes alive whereas with this the vm is allowed to collect them -- whether it does so often or rarely is up to it.

### CODE REVIEW

**Review by ahe-humio — APPROVED** [review]
This looks reasonable, however, @mortengrouleff mentioned that we should already be loading regular expression via class loader.

Given that, my understanding is that this is a cleanup, but not strictly necessary. Am I understanding this correctly?

**Comment by ahe-humio** [comment]
I've added Kresten as a reviewer. I don't think you have to wait for his approval, but its nice to keep him in the loop since he's done a lot of work on this.

**Comment by plesner** [comment]
I don't see how it's possible for a caller to change how the classes are loaded. The static shared loader looks pretty hardcoded. So as far as I know this is a fix not a cleanup.

**Review by mortengrouleff — APPROVED** [review]
LGTM.
Oh, that loader was static where it should not have been...
But I'm fine with the loader being only where it is used, that's even better.

**Comment by ahe-humio** [comment]
> I don't see how it's possible for a caller to change how the classes are loaded. The static shared loader looks pretty hardcoded. So as far as I know this is a fix not a cleanup.

It's possible to load Jitrex itself using another class loader. Each such class loader would have its own version of the static variable. 

I thought that's what we're doing, but perhaps I misunderstood something.

**Pull request merged**
